### PR TITLE
Align the error message of Cast with Presto

### PR DESCRIPTION
**Pull request by jinchengchenghh**

### CODE REVIEW

**Comment by jinchengchenghh** [comment]
Can you helo review this PR? Thanks! @mbasmanova

File: velox/type/Type.h [hunk 1]
```diff
@@ -638,6 +638,7 @@ class DecimalType : public ScalarType<KIND> {
   static_assert(KIND == TypeKind::BIGINT || KIND == TypeKind::HUGEINT);
   static constexpr uint8_t kMaxPrecision = KIND == TypeKind::BIGINT ? 18 : 38;
   static constexpr uint8_t kMinPrecision = KIND == TypeKind::BIGINT ? 0 : 19;
+  static constexpr uint8_t kMaxScale = kMaxPrecision;
```
Comment (mbasmanova) [inline velox/type/Type.h:641]: Unrelated change?

**Review by mbasmanova — APPROVED** [review]
@jinchengchenghh Thanks.

**Comment by mbasmanova** [comment]
@jinchengchenghh This change breaks Prestissimo test:

```
[ RUN      ] TestTypeSignature.decimalType
fbcode/github/presto-trunk/presto-native-execution/presto_cpp/main/types/tests/TypeSignatureTest.cpp:403: Failure
Expected equality of these values:
  parseTypeSignature("decimal(10, 5)")->toString()
    Which is: "DECIMAL(10, 5)"
  "DECIMAL(10,5)"
[  FAILED  ] TestTypeSignature.decimalType (2 ms)
```

CC: @aditi-pandit @majetideepak @spershin 

I'm not sure if there is a way to make this change backwards-compatible. Assuming not, one option would be to first update TypeSignatureTest to allow both DECIMAL(10, 5) and DECIMAL(10,5), then land this change, then update TypeSignatureTest to remove DECIMAL(10,5).

**Comment by jinchengchenghh** [comment]
I suppose not, we should do as latter solution. Does TestTypeSignature in your private repo? Can any one help update it? @mbasmanova

**Comment by mbasmanova** [comment]
> I suppose not, we should do as latter solution. Does TestTypeSignature in your private repo? Can any one help update it? @mbasmanova

This code is in Presto repo: https://github.com/prestodb/presto/blob/master/presto-native-execution/presto_cpp/main/types/tests/TypeSignatureTest.cpp

**Comment by aditi-pandit** [comment]
> @jinchengchenghh This change breaks Prestissimo test:
> 
> ```
> [ RUN      ] TestTypeSignature.decimalType
> fbcode/github/presto-trunk/presto-native-execution/presto_cpp/main/types/tests/TypeSignatureTest.cpp:403: Failure
> Expected equality of these values:
>   parseTypeSignature("decimal(10, 5)")->toString()
>     Which is: "DECIMAL(10, 5)"
>   "DECIMAL(10,5)"
> [  FAILED  ] TestTypeSignature.decimalType (2 ms)
> ```
> 
> CC: @aditi-pandit @majetideepak @spershin
> 
> I'm not sure if there is a way to make this change backwards-compatible. Assuming not, one option would be to first update TypeSignatureTest to allow both DECIMAL(10, 5) and DECIMAL(10,5), then land this change, then update TypeSignatureTest to remove DECIMAL(10,5).

@mbasmanova : Yes, update the TypeSignatureTest to allow both and then update the test to remove DECIMAL(10,5) sounds good.

**Comment by jinchengchenghh** [comment]
https://github.com/prestodb/presto/pull/20710 has been merged, could we merge this one? @mbasmanova

**Comment by mbasmanova** [comment]
@jinchengchenghh Please, rebase. I'll try to merge then.
